smart/forward_model/model_fit.py

```diff
@@ -175,7 +175,12 @@ def makeModel(teff, logg=5, metal=0, vsini=1, rv=0, tell_alpha=1.0, airmass=1.0,
         stellar_model.wave += wave_offset
         if binary:
             model2.wave = stellar_model.wave
-
+    
+    # super-resampling
+    f_model = interp1d(x=model.wave, y=model.flux)
+    model.wave = np.linspace(min(model.wave), max(model.wave), len(data.wave))
```

Comment:
Is it always True that we are sampling wavelength in linear space?
Do we have to do this for every model now?

Reply:
Probably I should add a judgement statement: resample only when the model's resolution is lower than the data's. A value error would be raised whenever I pass super-sampled data exceeding the original resolution into the function. In my case the spacing is smaller than the lowest resolution of the model itself, so it doesn't matter which way it's resampling anymore.

Reply:
I'm not sure about the first question. I believe we were always sampling wavelength in linear space. We could do it in velocity space too, but I'm not sure there's an advantage to that. If we keep the native sampling of the models (which I vote for), then we need to increase the resolution of the telluric model. One simple fix is to just generate supersampled models that can be read in once instead of supersampling each step.

Reply:
The wavelength solution of the spectral data is typically nonlinear from pixel to wavelength space (although in a narrow spectral order this is roughly the same). I would suggest adding an option for this like ```supersample_model=True``` (default = False). I don't feel comfortable when doing unnecessary supersampling. This should mostly dependent on which models we are using.

I thought we just have to downsample the theoretical models below the telluric models, but this line would make the resolution the same as the data? Also note that in the next few lines, it performs another resample ```integralResample``` again.

Reply:
Yes, I think this is very reasonable to implement going forward.